Infra and Deployment for ECS Fargate

trial-danswer opened the pull request:
## Description

[Provide a brief description of the changes in this PR]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes]

## Backporting (check the box to trigger backport action)

Note: You have to check that the action passes, otherwise resolve the conflicts manually and tag the patches.

- [ ] This PR should be backported (make sure to check that the backport attempt succeeds)
- [ ] [Optional] Override Linear Check

Weves commented on deployment/data/nginx/app.conf.template:
```diff
@@ -70,21 +70,3 @@ server {
     }
 }
-
-server {
```
we probably need to keep this around

trial-danswer commented:
I've updated the PR with changes to include horizontal auto-scaling of tasks based on cpu and memory for api-backend, background, model-servers, and web-server.  Additionally, I've updated the templates to accept parameters for task memory, cpu, and desired count. Lastly, I set the default memory & cpu for tasks based on approx values from the Kubernetes helm-charts.  The ECS Fargate mem/cpu values are based on the following table: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html

Weves commented on deployment/aws_ecs_fargate/cloudformation/services/onyx_services_parameters.json:
```diff
@@ -0,0 +1,14 @@
+	"ParameterValue": "subnet-0af372d676f65b287,subnet-06721b79e2e51d22c"
+},
+{
+	"ParameterKey": "VpcID",
```
similarly could there be a single place to configure resources across all the services?

trial-danswer replied:
The parameters.json file passed to aws cloudformation deploy must match the parameters defined in the specific CloudFormation template being deployed. If a parameter is present in the JSON file but not defined in the template, CloudFormation will reject it. 

Possible to write a script that would parse out a single master parameter file...but that would become complex.

trial-danswer commented on deployment/aws_ecs_fargate/cloudformation/services/onyx_services_parameters.json:
```diff
@@ -0,0 +1,14 @@
+[
+{
```
I had to separate the nginx parameters into a separate parameters.json file due to the fact that parameter file must match what is defined in the cloudformation template.  And the nginx cloudformation template has unqiue parameters defined unlike the other onyx services.

trial-danswer commented on deployment/data/nginx/app.conf.template.use-cloudtls:
```diff
@@ -0,0 +1,69 @@
+# Log format to include request latency
```
I merged it into app.conf.template.dev.  Prior to adding the env variables to run-nginx.sh I had to keep it separated..but now it can be a single conf.
